Cython/Compiler/Nodes.py

```diff
@@ -5310,6 +5307,16 @@ def analyse_declarations(self, env):
         self.scope = scope = self.entry.type.scope
         if scope is not None:
             scope.directives = env.directives
+            if "dataclasses.dataclass" in env.directives:
+                is_frozen = False
+                dataclass_directives = env.directives["dataclasses.dataclass"]
+                if dataclass_directives:
+                    frozen_directive = dataclass_directives[1].get('frozen', None)
+                    is_frozen = frozen_directive and frozen_directive.is_literal and frozen_directive.value
```

Comment:
We probably already know the node's `constant_result` at this point. (You can ask `has_constant_result()`, but there's hopefully some prior piece of validation code that makes sure that we *do* know the constant result of this flag at compile time.)

Reply:
I've changed to using `constant_result`. We do currently enforce `is_literal` on these (https://github.com/cython/cython/blob/5ddbe05d419d48161e5938e8de227d41913312ee/Cython/Compiler/Dataclass.py#L181) but I guess `constant_result` is one less thing to change if we change that rule later

Reply:
Don't think `constant_result` will work. I think directives are taken out of the visited tree before the `ConstantFolding` step so `is_literal` is probably the only way to do it.